File: C#/Veiculo/Veiculo/objetos/Aviao.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Veiculo.objetos
{
    class Aviao : Veiculo
    {
        private String tipo;
        private String uso;
        public Aviao(string marca, string modelo, string placa, float km, bool isLigado, int litrosCombustivel, int velocidade, double preco, String tipo, String uso) : base(marca, modelo, placa, km, isLigado, litrosCombustivel, velocidade, preco)
        {
            this.Tipo = tipo;
            this.Uso = uso;
        }

        public string Tipo { get => tipo; set => tipo = value; }
        public string Uso { get => uso; set => uso = value; }


		public void abastecer(String qtdLitros)
		{
			if (LitrosCombustivel >= 1000)
			{
				Console.WriteLine("Combustivel cheio");
			}
			else
			{
				if (LitrosCombustivel + int.Parse(qtdLitros) >= 1000)
				{
					Console.WriteLine("abastecido 100%");
					LitrosCombustivel = 1000;
				}
				else
				{
					LitrosCombustivel += int.Parse(qtdLitros);
				}

			}
		}



		
	public void acelerar()
		{
			if (IsLigado == true)
			{
				Velocidade += 1000;
				LitrosCombustivel -= 20;
			}
			else if (LitrosCombustivel < 1)
			{
				Console.WriteLine("impossivel acelerar");
			}
			else
			{
				Console.WriteLine("o veiculo está desligado");
			}
		}

		public override bool Equals(object obj)
		{
			return base.Equals(obj);
		}

		public override int GetHashCode()
		{
			return base.GetHashCode();
		}

		public override string ToString()
		{
			return base.ToString();
		}
	}
}

File: aula 3/com.br.gft/src/com/objetos/Aviao.java
package com.objetos;

public class Aviao extends Veiculo {
	public Aviao(String marca, String modelo, String placa, float km, boolean isLigado, int litrosCombustivel,
			int velocidade, double preco, String tipo, String uso) {
		super(marca, modelo, placa, km, isLigado, litrosCombustivel, velocidade, preco);
		this.tipo = tipo;
		this.uso = uso;
	}

	private String tipo;
	private String uso;
	

	public String getTipo() {
		return tipo;
	}

	public void setTipo(String tipo) {
		this.tipo = tipo;
	}

	public String getUso() {
		return uso;
	}

	public void setUso(String uso) {
		this.uso = uso;
	}
	
	
	public void abastecer(String qtdLitros) {
		if (litrosCombustivel >= 1000) {
			System.out.println("Combustivel cheio");
		} else {
			if (litrosCombustivel + Integer.parseInt(qtdLitros) >= 1000) {
				System.out.println("abastecido 100%");
				litrosCombustivel = 1000;
			} else {
				litrosCombustivel += Integer.parseInt(qtdLitros);
			}

		}
	}
	
	
	
	@Override
	public void acelerar() {
		if (this.isLigado == true) {
			velocidade += 1000;
			litrosCombustivel -= 20;
		} else if (this.litrosCombustivel < 1) {
			System.out.println("impossivel acelerar");
		} else {
			System.out.println("o veiculo está desligado");
		}
	}

}
